e2e/single-cluster/imagescan_test.go

```diff
@@ -92,3 +68,211 @@ var _ = Describe("Image Scan", Label("infra-setup"), func() {
+			// store number of fleet controller restarts to compare later
+			fleetControllerRestarts := getFleetControllerRestarts(k)
+			newTag := "0.0.0-50"
+			previousImageTag := imageTag
+			imageTag = tagAndPushImage("k8s.gcr.io/pause", image, newTag)
+
+			// the scan time interval is 5 seconds.
+			// we check for 10 seconds so we're sure that the image has been scanned and the controller didn't crash
+			Consistently(func() int {
+				return getFleetControllerRestarts(k)
+			}, 10*time.Second, 1*time.Second).Should(
+				Equal(fleetControllerRestarts))
+
+			By("checking the bundle has the original image tag")
+			Eventually(func() string {
+				out, _ := k.Get("bundles", "imagescan-examples", "-o", "yaml")
+				return out
+			}).Should(
+				MatchRegexp(fmt.Sprintf(`image: %s # {"\$imagescan": "test-scan"}`, previousImageTag)))
+		})
+	})
+	AfterEach(func() {
+		_, _ = k.Delete("gitrepo", "imagescan")
+		_, _ = k.Delete("secret", "git-auth")
+	})
+})
+
+func getFleetControllerRestarts(k kubectl.Command) int {
+	out, err := k.Namespace("cattle-fleet-system").Get("pods", "-l", "app=fleet-controller", "--no-headers")
+	Expect(err).NotTo(HaveOccurred())
+	items := strings.Fields(out)
+	// it has more columns when showing restarts
+	Expect(len(items)).Should(BeNumerically(">=", 5))
+	// restarts is shown at position 4
+	n, err := strconv.Atoi(items[3])
+	Expect(err).NotTo(HaveOccurred())
+	return n
+}
+
+func setupRepo(k kubectl.Command, tmpdir, clonedir, repoDir string) *git.Repository {
+	// Create git secret
+	out, err := k.Create(
+		"secret", "generic", "git-auth", "--type", "kubernetes.io/basic-auth",
+		"--from-literal=username="+os.Getenv("GIT_HTTP_USER"),
+		"--from-literal=password="+os.Getenv("GIT_HTTP_PASSWORD"),
+	)
+	Expect(err).ToNot(HaveOccurred(), out)
+
+	addr, err := githelper.GetExternalRepoAddr(env, port, repoName)
+	Expect(err).ToNot(HaveOccurred())
+	gh := githelper.NewHTTP(addr)
+	gh.Branch = "imagescan"
+
+	repo, err := gh.Create(clonedir, repoDir, "examples")
+	Expect(err).ToNot(HaveOccurred())
+
+	// Build git repo URL reachable _within_ the cluster, for the GitRepo
+	host, err := githelper.BuildGitHostname(env.Namespace)
+	Expect(err).ToNot(HaveOccurred())
+
+	inClusterRepoURL := gh.GetInClusterURL(host, port, repoName)
+
+	gitrepo := path.Join(tmpdir, "gitrepo.yaml")
+	err = testenv.Template(gitrepo, testenv.AssetPath("imagescan/imagescan.yaml"), struct {
+		Repo   string
+		Branch string
+	}{
+		inClusterRepoURL,
+		gh.Branch,
+	})
+	Expect(err).ToNot(HaveOccurred())
+
+	out, err = k.Apply("-f", gitrepo)
+	Expect(err).ToNot(HaveOccurred(), out)
+	return repo
+}
+
+func tagAndPushImage(baseImage, image, tag string) string {
+	imageTag := fmt.Sprintf("%s:%s", image, tag)
+	// tag the image and push it to ttl.sh
+	cmd := exec.Command("docker", "tag", baseImage, imageTag)
+	err := cmd.Run()
+	Expect(err).ToNot(HaveOccurred())
+	// push the image to ttl.sh
+	cmd = exec.Command("docker", "push", imageTag)
```

Comment:
Out of curiosity, do we know the default expiration time of `ttl.sh` images? I've tried to reach ttl.sh to check that, but am getting a blank page, which did not use to be the case :thinking: .

Reply:
I'm not 100% sure [this](https://www.civo.com/learn/ttl-sh-your-anonymous-and-ephemeral-docker-image-registry) is a reliable source, but it states that the default expiration time is 1h and maximum is 24h. 

As we need to specify pre-release tags we can't use the tag to specify the expiration time, so I assume they expire in 1h.

Reply:
Nice, thanks for checking. That makes sense, 1h should be enough for our use case.